sopel/coretasks.py

```diff
@@ -120,15 +152,28 @@ def auth_after_register(bot):
 
 
 def _execute_perform(bot):
-    """Execute commands specified to perform on IRC server connect."""
+    """Execute commands specified to perform on IRC server connect.
+
+    This function executes the list of commands that can be found in the
+    ``core.commands_on_connect`` setting. It automatically replace any
+    ``$nickname`` placeholder in the command with the bot's configured nick.
+    """
     if not bot.connection_registered:
         # How did you even get this command, bot?
         raise Exception('Bot must be connected to server to perform commands.')
 
-    LOGGER.debug('{} commands to execute:'.format(len(bot.config.core.commands_on_connect)))
-    for i, command in enumerate(bot.config.core.commands_on_connect):
+    commands = bot.config.core.commands_on_connect
+    count = len(commands)
+
+    if not count:
+        LOGGER.info('No custom command to execute.')
+        return
+
+    LOGGER.info('Executing %d custom commands.', count)
+    for i, command in enumerate(commands, 1):
         command = command.replace('$nickname', bot.config.core.nick)
-        LOGGER.debug(command)
+        LOGGER.debug(
+            'Executing custom command [%d/%d]: %s', i, count, command)
         bot.write((command,))
```

Comment:
This mini-refactor belongs in its own commit.

Reply:
Yes... x)

Reply:
Which I've done!